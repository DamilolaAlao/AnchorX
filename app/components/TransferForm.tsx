import * as React from 'react';
import { Alert } from 'react-native'
import { Container, Content, Form, Item, Input, Label, Button, Text, Spinner } from 'native-base';
import { styles as s } from "react-native-style-tachyons";

import { Transaction } from '../Types'

interface Props {
  send: (amount: string) => Promise<Transaction>
}

interface State {
  amount: string
  sending: boolean
}

class TransferForm extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      amount: '',
      sending: false
    }
  }

  async send(): Promise<Transaction> {
    const { amount } = this.state

    this.setState({
      sending: true
    })

    const transaction = await this.props.send(amount)

    this.setState({
      sending: false
    })

    return transaction
  }

  render() {
    const { amount, sending } = this.state

    return (
      <Container>
        <Content scrollEnabled={false}>
          <Form>
            <Item floatingLabel>
              <Label>Amount</Label>
              <Input
                onChangeText={(amount) => this.setState({ amount })}
                value={amount}
                keyboardType={'decimal-pad'}
                autoFocus={true}
                autoCorrect={false}
                autoCapitalize={"none"}
              />
            </Item>
            {!sending && (
               <Button full style={[s.mt4]} onPress={() => this.send()}>
                 <Text>Send</Text>
               </Button>
            )}
            {sending && <Spinner color="blue" />}
          </Form>
        </Content>
      </Container>
    );
  }
}

export default TransferForm
