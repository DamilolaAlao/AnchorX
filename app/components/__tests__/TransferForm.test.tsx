import * as React from 'react'
import TransferForm from '../TransferForm'

import * as renderer from 'react-test-renderer'

const send = (amount: string) => {
  return new Promise.resolve({
    id: '1234'
  })
}

jest.useFakeTimers()

it('renders correctly', () => {
  const rendered = renderer.create(<TransferForm send={send} />).toJSON()

  expect(rendered).toMatchSnapshot()
});
