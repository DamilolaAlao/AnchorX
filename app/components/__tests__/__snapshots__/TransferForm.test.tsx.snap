// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#F5FCFF",
      },
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEnabled={false}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={Object {}}
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "flexDirection": "row",
              "marginLeft": 15,
              "marginTop": 15,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "left": 0,
                "opacity": 1,
                "paddingBottom": undefined,
                "paddingTop": undefined,
                "position": "absolute",
                "right": 0,
                "top": 18,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#575757",
                  "fontSize": 17,
                  "left": 0,
                  "paddingRight": 5,
                  "top": 6,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 17,
                  }
                }
              >
                Amount
              </Text>
            </Text>
          </View>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            autoFocus={true}
            editable={true}
            keyboardType="decimal-pad"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#575757"
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "marginLeft": null,
                "paddingBottom": 7,
                "paddingLeft": null,
                "paddingRight": 5,
                "paddingTop": 3,
                "top": 10,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
            value=""
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#007aff",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 0,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            Send
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
